src/api/app/models/kiwi/image.rb

```diff
@@ -1,181 +1,183 @@
-  </preferences>
-</image>
-'.freeze
-
-  #### Self config
-
-  #### Attributes
-
-  #### Associations macros (Belongs to, Has one, Has many)
-  has_one :package, foreign_key: 'kiwi_image_id', class_name: '::Package', dependent: :nullify, inverse_of: :kiwi_image
-  has_many :repositories, -> { order(order: :asc) }, dependent: :destroy, index_errors: true
-  has_many :package_groups, -> { order(:id) }, dependent: :destroy, index_errors: true
-  has_many :kiwi_packages, -> { where(kiwi_package_groups: { kiwi_type: Kiwi::PackageGroup.kiwi_types[:image] }) },
-           through: :package_groups, source: :packages, inverse_of: :kiwi_image
-
-  #### Callbacks macros: before_save, after_save, etc.
-
-  #### Scopes (first the default_scope macro if is used)
-
-  #### Validations macros
-  validates :name, presence: true
-  validate :check_use_project_repositories
-  validate :check_package_groups
-  accepts_nested_attributes_for :repositories, allow_destroy: true
-  accepts_nested_attributes_for :package_groups, allow_destroy: true
-  accepts_nested_attributes_for :kiwi_packages, allow_destroy: true
-
-  #### Class methods using self. (public and then private)
-
-  #### To define class methods as private use private_class_method
-  #### private
-
-  #### Instance methods (public and then protected/private)
-
-  #### Alias of methods
-  def self.build_from_xml(xml_string, md5)
-    xml = Xmlhash.parse(xml_string)
-    return new(name: 'New Image: Please provide a name', md5_last_revision: md5) if xml.blank?
-    new_image = new(name: xml['name'], md5_last_revision: md5)
-
-    repositories = [xml["repository"]].flatten.compact
-    new_image.use_project_repositories = repositories.any? { |repository| repository['source']['path'] == 'obsrepositories:/' }
-    repositories.reject{ |repository| repository['source']['path'] == 'obsrepositories:/' }.each.with_index(1) do |repository, index|
-      attributes = {
-        repo_type:   repository['type'],
-        source_path: repository['source']['path'],
-        priority:    repository['priority'],
-        order:       index,
-        alias:       repository['alias'],
-        replaceable: repository['status'] == 'replaceable',
-        username:    repository['username'],
-        password:    repository['password']
-      }
-      attributes['imageinclude'] = repository['imageinclude'] == 'true' if repository.key?('imageinclude')
-      attributes['prefer_license'] = repository['prefer-license'] == 'true' if repository.key?('prefer-license')
-
-      new_image.repositories.build(attributes)
-    end
+module Kiwi
+  class Image < ApplicationRecord
+    #### Includes and extends
+
+    #### Constants
+    DEFAULT_KIWI_BODY = '<?xml version="1.0" encoding="utf-8"?>
+  <image schemaversion="6.2" name="suse-13.2-live">
+    <description type="system">
+    </description>
+    <preferences>
+      <type image="oem" primary="true" boot="oemboot/suse-13.2"/>
+    </preferences>
+  </image>
+  '.freeze
```

Comment:
@ChrisBr The whole xml snippet is a string and shouldn't be indented. Otherwise the formatting of the xml is wrong and breaks our tests.
I did the same mistake when trying to use a [modularize](#3970) this;-)

Reply:
So means you already tried this and it didn't work?